tools/generate-interface.sh

```diff
@@ -0,0 +1,48 @@
+#! /bin/bash
+# generate-interface.sh generates the interface for the Payments contract
+
+set -euo pipefail
+
+# Check for required tools
+for cmd in forge jq npm; do
+  if ! command -v $cmd &> /dev/null; then
+    echo "Error: '$cmd' is not installed or not in PATH." >&2
+    exit 1
+  fi
+done
+
+# Build the contracts
+echo "Building contracts..."
+forge build --force > /dev/null 2>&1
```

Comment:
can we capture errors? maybe something like:

```suggestion
forge build --force 2>&1 | grep -Ei "^error" || true
```

(though I'm not too familiar with forge to know what errors consistently look like so maybe check?)

Reply:
or .. just let all the output past, is it really too much?